### ISSUE TITLE
ResourceDictionary.ThemeDictionaries isn't supported

### ISSUE THREAD

**Issue by davidjohnoliver** [unoplatform/uno#935]
## Current behavior

Resources inside ThemeDictionaries are ignored. Runtime error if these resources are referenced.

## Expected behavior

We need to think about how to handle dark/light/high-contrast on non-Windows platforms, but at a bare minimum the default resources should be discovered and available.

## Minimal reproduction of the problem with instructions

```xaml
	<UserControl.Resources>
		<ResourceDictionary>
			<ResourceDictionary.ThemeDictionaries>
				<ResourceDictionary x:Key="Default">
					<Style x:Key="BodyTextBlockMediumStyle"
                           BasedOn="{StaticResource BodyTextBlockStyle}"
                           TargetType="TextBlock">
						<Setter Property="Foreground" Value="{ThemeResource SystemControlForegroundBaseMediumBrush}"/>
					</Style>
				</ResourceDictionary>
				<ResourceDictionary x:Key="Light">
					<Style x:Key="BodyTextBlockMediumStyle"
                           BasedOn="{StaticResource BodyTextBlockStyle}"
                           TargetType="TextBlock">
						<Setter Property="Foreground" Value="{ThemeResource SystemControlForegroundBaseMediumBrush}"/>
					</Style>
				</ResourceDictionary>
				<ResourceDictionary x:Key="HighContrast">
					<Style x:Key="BodyTextBlockMediumStyle"
                           BasedOn="{StaticResource BodyTextBlockStyle}"
                           TargetType="TextBlock"/>
				</ResourceDictionary>
			</ResourceDictionary.ThemeDictionaries>
...
```

## Environment

Nuget Package: Uno.UI

Package Version(s): 1.45.0-dev.1581

Affected platform(s):
- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [ ] 2017 (version: )
- [ ] 2019 (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )

**Comment by jeromelaban** [comment]
This issue will enable support for https://github.com/unoplatform/calculator/issues/161

**Comment by carldebilly** [comment]
Implemented by PR #1188
